src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -496,72 +521,95 @@ func (c controller) postInstallConfigs(ctx context.Context) (error, map[string]i
 	return nil, nil
 }
 
-func (c controller) waitForOLMOperators(ctx context.Context) error {
+func (c *controller) CancelFinalizingStage(stage models.FinalizingStage) {
+	c.stagesCancellers.cancelForStage(stage)
+}
+
+func (c *controller) finalizingStageCtx(parentCtx context.Context, stage models.FinalizingStage) context.Context {
```

Comment:
Can you provide some explanation for this one?Not sure i understand

Reply:
There are some stages (today OLM stages), that timeout does not cause installation failure. Instead, the installation proceeds.
To accomplish it, when the controller monitors the cluster, it finds out if the current stage was timed out.
If it was timed out, the context us cancelled.
Since the context is also used by many others in the controller, a context for this stage was created separately and also can be cancelled separately.  This cancellation will not effect any other using the main context.